subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -651,17 +786,163 @@ static int engine_add_observer(struct lwm2m_message *msg,
 	return 0;
 }
 
-static void remove_observer_from_list(struct lwm2m_ctx *ctx, sys_snode_t *prev_node,
-				      struct observe_node *obs)
+static int engine_path_list_size(sys_slist_t *lwm_path_list)
 {
-	char buf[LWM2M_MAX_PATH_STR_LEN];
+	int list_size = 0;
+	struct lwm2m_obj_path_list *entry;
 
-	LOG_DBG("Removing observer %p for path %s", obs, lwm2m_path_log_strdup(buf, &obs->path));
+	SYS_SLIST_FOR_EACH_CONTAINER (lwm_path_list, entry, node) {
+		list_size++;
+	}
+	return list_size;
+}
 
-	if (ctx->observe_cb) {
-		ctx->observe_cb(LWM2M_OBSERVE_EVENT_OBSERVER_REMOVED, &obs->path, NULL);
+static bool engine_compare_obs_path_list(sys_slist_t *obs_path_list, sys_slist_t *path_list,
+					 int list_length)
+{
+	sys_snode_t *obs_ptr, *comp_ptr;
+	struct lwm2m_obj_path_list *obs_path, *comp_path;
+
+	obs_ptr = sys_slist_peek_head(obs_path_list);
+	comp_ptr = sys_slist_peek_head(obs_path_list);
+	while (list_length--) {
+		obs_path = (struct lwm2m_obj_path_list *) obs_ptr;
+		comp_path = (struct lwm2m_obj_path_list *) comp_ptr;
+		if (memcmp(&obs_path->path, &comp_path->path, sizeof(struct lwm2m_obj_path))) {
```

Comment:
I just wonder, if two observations are made with the same paths, but in different order shouldn't this also be detected as a duplicate? Or do we sort the path list at some point?

Reply:
We sort path at RX already. So thats why I return earlier